src/Nethermind/Nethermind.Taiko/Rpc/TaikoEngineRpcModule.cs

```diff
@@ -0,0 +1,331 @@
+    {
+        return base.engine_newPayloadV2(executionPayload);
+    }
+
+    public Task<ResultWrapper<ForkchoiceUpdatedV1Result>> engine_forkchoiceUpdatedV3(ForkchoiceStateV1 forkchoiceState, TaikoPayloadAttributes? payloadAttributes = null)
+    {
+        return base.engine_forkchoiceUpdatedV3(forkchoiceState, payloadAttributes);
+    }
+
+    public Task<ResultWrapper<PayloadStatusV1>> engine_newPayloadV3(TaikoExecutionPayloadV3 executionPayload, byte[]?[] blobVersionedHashes, Hash256? parentBeaconBlockRoot)
+    {
+        return base.engine_newPayloadV3(executionPayload, blobVersionedHashes, parentBeaconBlockRoot);
+    }
+
+    public ResultWrapper<PreBuiltTxList[]?> taikoAuth_txPoolContent(Address beneficiary, UInt256 baseFee, ulong blockMaxGasLimit,
+         ulong maxBytesPerTxList, Address[]? localAccounts, int maxTransactionsLists)
+    {
+        IEnumerable<KeyValuePair<AddressAsKey, Transaction[]>> pendingTxs =
+            txPool.GetPendingTransactionsBySender();
+
+        if (localAccounts is not null)
+        {
+            pendingTxs = pendingTxs.OrderBy(txs => !localAccounts.Contains(txs.Key.Value));
+        }
+
+        Transaction[] txQueue = pendingTxs.SelectMany(txs => txs.Value).Where(tx => !tx.SupportsBlobs && tx.CanPayBaseFee(baseFee)).ToArray();
+
+        BlockHeader? head = blockFinder.Head?.Header;
+
+        if (txQueue.Length is 0 || head?.StateRoot is null)
+        {
+            return ResultWrapper<PreBuiltTxList[]?>.Success([]);
+        }
+
+        IReadOnlyTxProcessorSource readonlyTxProcessingEnv = readOnlyTxProcessingEnvFactory.Create();
+        using IReadOnlyTxProcessingScope scope = readonlyTxProcessingEnv.Build(head.StateRoot);
+
+        return ResultWrapper<PreBuiltTxList[]?>.Success(ProcessTransactions(scope.TransactionProcessor, scope.WorldState, new BlockHeader(
+                head.Hash!,
+                Keccak.OfAnEmptySequenceRlp,
+                beneficiary,
+                UInt256.Zero,
+                head!.Number + 1,
+                (long)blockMaxGasLimit,
+                head.Timestamp + 1,
+                [])
+        {
+            TotalDifficulty = 0,
+            BaseFeePerGas = baseFee,
+            StateRoot = head.StateRoot,
+            IsPostMerge = true,
+        }, txQueue, maxTransactionsLists, maxBytesPerTxList));
+    }
+
+
+
+    private PreBuiltTxList[] ProcessTransactions(ITransactionProcessor txProcessor, IWorldState worldState, BlockHeader blockHeader, Transaction[] txSource, int maxBatchCount, ulong maxBytesPerTxList)
+    {
+        if (txSource.Length is 0 || blockHeader.StateRoot is null)
+        {
+            return [];
+        }
+
+        List<PreBuiltTxList> Batches = [];
+
+        void CommitAndDisposeBatch(Batch batch)
+        {
+            Batches.Add(new PreBuiltTxList(batch.Transactions.Select(tx => new LegacyTransactionForRpc(tx)).ToArray(),
```

Comment:
To fix the compile error, I replaced the abstract `TransactionForRpc` with `LegacyTransactionForRpc`. Initially, I tried with the `TransactionForRpc.FromTransaction()` method, but in `TxPoolContentListsTests`, it expects the `Input` property of tx.

@flcl42 @emlautarom1 I believe that test can be revised.

Reply:
Might be cool to make a test tx type with TestTransactionId or so. But for now I'll simply use OfType in the test